src/Kros.MediatR.Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Kros.MediatR.PostProcessor;
+using MediatR;
+using MediatR.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Kros.MediatR.Extensions
+{
+    /// <summary>
+    /// Extensions to scan for MediatR pipeline behaviors and registers them.
+    /// </summary>
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Scan for MediatR pipeline behaviors
+        /// which request implement <typeparamref name="TRequest"/> and response implement <typeparamref name="TResponse"/>.
+        /// </summary>
+        /// <typeparam name="TRequest">Request type.</typeparam>
+        /// <typeparam name="TResponse">Response type.</typeparam>
+        /// <param name="services">Service container.</param>
+        /// <exception cref="InvalidOperationException">When number of implementation
+        /// <typeparamref name="TRequest"/> and <typeparamref name="TResponse"/> are different.</exception>
+        public static IServiceCollection AddPipelineBehaviorsForRequest<TRequest, TResponse>(this IServiceCollection services)
+        {
+            var requestType = typeof(TRequest);
+            var responseType = typeof(TResponse);
+            var pipeLineType = typeof(IPipelineBehavior<,>);
+            (var requests, var responses) = GetTypes(requestType, responseType);
+
+            Type MakeGenericType(Type type, int i)
+                => type.MakeGenericType(requests[i], responses[i]);
+
+            IEnumerable<Type> pipelineBehaviours = GetPipelineBehaviours(requestType, pipeLineType);
```

Comment:
Inconsistent naming. Here you are using `Behaviours` and above `Behaviors`. Both are correct but inconsistent.